src/components/SearchBox/index.js

```diff
@@ -6,6 +6,7 @@ export default class SearchBox extends React.Component {
     placeholder: React.PropTypes.string.isRequired,
     onChange: React.PropTypes.func.isRequired,
     onValidate: React.PropTypes.func.isRequired,
+    showSearchIcon: React.PropTypes.bool,
```

Comment:
I think we should not have an option to decide to show the search icon or not. Best practice is to have one (cc @Shipow), so lets always have it and if the user really wants to hide it then it can be done with CSS.